src/config/enterprise.rs

```diff
@@ -375,26 +375,25 @@ fn setup_logs_reporting(
         .as_ref()
         .map_or("".to_string(), |tags| convert_tags_to_vrl(tags, false));
 
+    let configuration_key = &datadog.configuration_key;
+    let vector_version = crate::vector_version();
+    let build_arch = built_info::TARGET_ARCH;
+    let build_os = built_info::TARGET_OS;
+    let build_vendor = built_info::TARGET_VENDOR;
     let tag_logs = RemapConfig {
         source: Some(format!(
             r#"
-            .version = "{}"
-            .configuration_key = "{}"
             .ddsource = "vector"
             .vector = {{
-                "version": "{}",
-                "arch": "{}",
-                "os": "{}",
-                "vendor": "{}"
+                "configuration_key": "{configuration_key}",
+                "version_hash": "{version_hash}",
+                "version": "{vector_version}",
+                "arch": "{build_arch}",
+                "os": "{build_os}",
+                "vendor": "{build_vendor}"
```

Comment:
I wonder if we should add these to the internal metrics too, in the future. I could see users being interested in viewing the metrics by, e.g., OS.